assets/js/common/ApiKeySettingsModal/ApiKeySettingsModal.jsx

```diff
@@ -0,0 +1,176 @@
+import {
+  addDays,
+  addMonths,
+  addYears,
+  format,
+  parseISO,
+  setHours,
+  setMinutes,
+} from 'date-fns';
+import React, { useState } from 'react';
+import { noop, truncate } from 'lodash';
+import { EOS_CONTENT_COPY, EOS_INFO_OUTLINED } from 'eos-icons-react';
+import Button from '@common/Button';
+import Modal from '@common/Modal';
+import Input from '@common/Input';
+import Label from '@common/Label';
+import Select from '@common/Select';
+import Switch from '@common/Switch';
+
+const normalizeExpiration = (expiration) =>
+  setMinutes(setHours(expiration, 0), 0);
+
+const availableTimeOptions = [
+  {
+    type: 'months',
+    timeGenerator: (quantity) => addMonths(new Date(), quantity),
+  },
+  {
+    type: 'days',
+    timeGenerator: (quantity) => addDays(new Date(), quantity),
+  },
+  {
+    type: 'years',
+    timeGenerator: (quantity) => addYears(new Date(), quantity),
+  },
+];
+
+function ApiKeySettingsModal({
+  open = false,
+  loading = false,
+  onSave = noop,
```

Comment:
What is `onSave`? when you click the `Generate` button?
Should we maybe renamed to `onGenerate` ?

Reply:
done